docs/devsite-help/client-lifecycle.md

```diff
@@ -87,6 +89,18 @@ Any properties explicitly set on the builder will not be requested from the serv
 if credentials are configured (in any form) on the builder, the `Build(IServiceProvider)` method will
 not request `ChannelCredentials`, `ICredential` or `GoogleCredential`.
 
+Additionally, in environments where Grpc.Net.Client will be used (e.g. .NET Core or .NET 5+) we recommend
+adding the `GrpcNetClientAdapter` to the dependency injection services, which allows logging and `HttpClient`
+construction to be configured in a consistent manner. For example:
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+builder.Services
+    .AddRazorPages()
+    .AddGrpcNetClientAdapter()
+    .AddExampleClient();
```

Comment:
Except that if you're using .NET Framework, you don't *want* to add the Grpc.Net.Client adapter.
I wondered about that myself - I suspect we can improve things over time, but let's wait until we get some more feedback.

Reply:
Yep, that's why I added that last line there about duplicating the transport selection logic, maybe we can call the method `MaybeAddDefaultClientAdatpter` and duplicate the detection logic here.

We still need it in the client builder though as people might use that directly instead of DI.